addon/index.js

```diff
@@ -1,27 +1,61 @@
 /* eslint-disable ember/new-module-imports */
 import Ember from 'ember';
-import { macroCondition, dependencySatisfies } from '@embroider/macros';
+import {
+  macroCondition,
+  dependencySatisfies,
+  importSync,
+} from '@embroider/macros';
 
-export const Checkbox = macroCondition(
-  dependencySatisfies('ember-source', '>= 3.27.0-beta.1')
-)
-  ? Ember._LegacyCheckbox.extend()
-  : Ember.Checkbox;
+let Checkbox;
```

Comment:
Maybe my personal preference, but nested ternary operators hard to read.

Another option could be to do smth like:
```js
export const Checkbox = macroCondition(
  dependencySatisfies('ember-source', '>= 3.27.0-beta.1')
)
  ? importSync('@ember/legacy-built-in-components/components/checkbox').default
  : Ember.Checkbox;
```
but I wasn't confident as this addon was using `Ember._LegacyCheckbox.extend()` for 3.27 and 3.28